From cd92c9974898cdbd0cf8bca38f9c502f0b3b3f39 Mon Sep 17 00:00:00 2001
From: ctang <ctang@cloudera.com>
Date: Mon, 25 Jul 2016 09:24:01 -0400
Subject: [PATCH 0706/1164] HIVE-14298: NPE could be thrown in HMS when an
 ExpressionTree could not be made from a filter
 (Chaoyu Tang, reviewed by Sergio Pena)

Change-Id: I38088006eab5a4c22c783dba91c07e9e3672488c
---
 .../hadoop/hive/metastore/MetaStoreDirectSql.java  |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java b/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java
index 0e8a595..14cd8d9 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java
@@ -971,6 +971,11 @@ private static String generateSqlFilter(Table table, ExpressionTree tree, List<O
         List<String> joins, boolean dbHasJoinCastBug, String defaultPartName, DB dbType)
             throws MetaException {
       assert table != null;
+      if (tree == null) {
+        // consistent with other APIs like makeExpressionTree, null is returned to indicate that
+        // the filter could not pushed down due to parsing issue etc
+        return null;
+      }
       if (tree.getRoot() == null) {
         return "";
       }
-- 
1.7.9.5

